test/integration/webhook/workload_test.go

```diff
@@ -260,5 +612,431 @@ var _ = ginkgo.Describe("Workload validating webhook", func() {
+					Name:               "ac1",
+					Message:            "old",
+					LastTransitionTime: metav1.NewTime(time.Now()),
+					PodSetUpdates:      []kueue.PodSetUpdate{{Name: "first", Labels: map[string]string{"foo": "bar"}}, {Name: "second"}},
+					State:              kueue.CheckStateReady,
+				})
+				return k8sClient.Status().Update(ctx, wl)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			ginkgo.By("Updating admission check state")
+			gomega.Eventually(func() error {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), wl)).To(gomega.Succeed())
+				workload.SetAdmissionCheckState(&wl.Status.AdmissionChecks, kueue.AdmissionCheckState{
+					Name:               "ac1",
+					Message:            "new",
+					LastTransitionTime: metav1.NewTime(time.Now()),
+					PodSetUpdates:      []kueue.PodSetUpdate{{Name: "first", Labels: map[string]string{"foo": "baz"}}, {Name: "second"}},
+					State:              kueue.CheckStateReady,
+				})
+				return k8sClient.Status().Update(ctx, wl)
+			}, util.Timeout, util.Interval).Should(testing.BeAPIError(testing.ForbiddenError))
+		})
+
+		ginkgo.It("should change other fields of admissionchecks when podSetUpdates is immutable", func() {
+			ginkgo.By("Creating a new Workload")
+			workload := testing.MakeWorkload(workloadName, ns.Name).
+				PodSets(
+					*testing.MakePodSet("first", 1).Obj(),
+					*testing.MakePodSet("second", 1).Obj(),
+				).AdmissionChecks(
+				kueue.AdmissionCheckState{
+					Name:          "ac1",
+					Message:       "old",
+					PodSetUpdates: []kueue.PodSetUpdate{{Name: "first", Labels: map[string]string{"foo": "bar"}}, {Name: "second"}},
+					State:         kueue.CheckStateReady,
+				}).Obj()
+			gomega.Expect(k8sClient.Create(ctx, workload)).Should(gomega.Succeed())
+
+			ginkgo.By("Updating the workload")
+			gomega.Eventually(func() error {
+				var newWL kueue.Workload
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(workload), &newWL)).To(gomega.Succeed())
+				newWL.Status.AdmissionChecks = []kueue.AdmissionCheckState{
+					{
+						Name:               "ac1",
+						Message:            "new",
+						LastTransitionTime: metav1.NewTime(time.Now()),
+						PodSetUpdates:      []kueue.PodSetUpdate{{Name: "first", Labels: map[string]string{"foo": "bar"}}, {Name: "second"}},
+						State:              kueue.CheckStateReady,
+					},
+				}
+				return k8sClient.Update(ctx, &newWL)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.It("updating priorityClassName before setting reserve quota for workload", func() {
+			ginkgo.By("Creating a new Workload")
+			workload := testing.MakeWorkload(workloadName, ns.Name).
+				Queue("q").
+				PriorityClass("test-class-1").PriorityClassSource(constants.PodPriorityClassSource).
+				Priority(10).Obj()
+			gomega.Expect(k8sClient.Create(ctx, workload)).Should(gomega.Succeed())
+
+			ginkgo.By("Updating the workload")
```

Comment:
```suggestion
			ginkgo.By("Checking the created workload")
```

Reply:
Description changed indicating the field that is being updated